control-plane-operator/controllers/hostedcontrolplane/routecm/reconcile.go

```diff
@@ -0,0 +1,239 @@
+package routecm
+
+import (
+	"fmt"
+	"path"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/common"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/kas"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	component "github.com/openshift/hypershift/support/controlplane-component"
+	"github.com/openshift/hypershift/support/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+)
+
+const (
+	ComponentName = "openshift-route-controller-manager"
+
+	ConfigMapName = "openshift-route-controller-manager-config"
+
+	configHashAnnotation       = "openshift-route-controller-manager.hypershift.openshift.io/config-hash"
+	servingPort          int32 = 8443
+)
+
+var _ component.DeploymentReconciler = &RouteControllerManagerReconciler{}
+
+type RouteControllerManagerReconciler struct {
+}
+
+func NewComponent() component.ControlPlaneComponent {
+	return &component.ControlPlaneWorkload{
+		DeploymentReconciler: &RouteControllerManagerReconciler{},
+		ResourcesReconcilers: []component.GenericReconciler{
+			component.NewReconcilerFor(&corev1.ConfigMap{}).
+				WithName(ConfigMapName).
+				WithReconcileFunction(ReconcileOpenShiftRouteControllerManagerConfig).
+				Build(),
+			component.NewReconcilerFor(&corev1.Service{}).
+				WithName(ComponentName).
+				WithReconcileFunction(ReconcileService).
+				WithPredicate(component.DisableIfAnnotationExist(hyperv1.DisableMonitoringServices)).
+				Build(),
+			component.NewReconcilerFor(&prometheusoperatorv1.ServiceMonitor{}).
+				WithName(ComponentName).
+				WithReconcileFunction(ReconcileServiceMonitor).
+				WithPredicate(component.DisableIfAnnotationExist(hyperv1.DisableMonitoringServices)).
+				Build(),
+		},
+
+		MultiZoneSpreadLabels: openShiftRouteControllerManagerLabels(),
+	}
+}
+
+// Name implements controlplanecomponent.DeploymentReconciler.
+func (r *RouteControllerManagerReconciler) Name() string {
+	return ComponentName
+}
+
+// ReconcileDeployment implements controlplanecomponent.DeploymentReconciler.
+func (r *RouteControllerManagerReconciler) ReconcileDeployment(cpContext component.ControlPlaneContext, deployment *appsv1.Deployment) error {
+	image := cpContext.ReleaseImageProvider.GetImage("route-controller-manager")
```

Comment:
For building the deployment reconciler, maybe we can use a builder pattern similar to what you have with the generic reconciler.

Reply:
makes sense, done!